rebasehelper/tests/plugins/test_spec_hooks.py

```diff
@@ -70,28 +112,59 @@ def test_escape_macros_spec_hook(self, spec_object):
         'new-version',
         'macro',
     ])
-    def test_replace_old_version_spec_hook(self, spec_object, replace_with_macro):
-        new_spec = spec_object.copy('new.spec')
-        new_spec.set_version('1.1.0')
-        ReplaceOldVersion.run(spec_object, new_spec, replace_old_version_with_macro=replace_with_macro)
+    @pytest.mark.parametrize('spec_attributes', [
+        {
+            'spec_content': dedent("""\
+                Version: 1.0.2
+                Source9: https://test.com/#/1.0/%{name}-hardcoded-version-1.0.2.tar.gz
+                Recommends: test > 1.0.2
+
+                %changelog
+                * Wed Apr 26 2017 Nikola Forró <nforro@redhat.com> - 1.0.2-34
+                - Update to 1.0.2
+                """),
+            'header': SimpleNamespace(version='1.0.2'),
+        }
+    ])
+    def test_replace_old_version_spec_hook(self, mocked_spec_object, replace_with_macro):
+        new_spec = copy.deepcopy(mocked_spec_object)
+        new_spec.header.version = '1.1.0'
+        new_spec.set_raw_tag_value('Version', '1.1.0')
+        ReplaceOldVersion.run(mocked_spec_object, new_spec, replace_old_version_with_macro=replace_with_macro)
+        # The spec is not saved due to mocking, refresh tags for the assertions
+        new_spec.tags = Tags(new_spec.spec_content, new_spec.spec_content)
+
         # Check if the version has been updated
-        test_source = [line for line in new_spec.spec_content.section('%package') if line.startswith('Source9')]
+        test_source = new_spec.get_raw_tag_value('Source9')
         assert test_source
         if replace_with_macro:
             expected_result = 'https://test.com/#/1.1/%{name}-hardcoded-version-%{version}.tar.gz'
         else:
             expected_result = 'https://test.com/#/1.1/%{name}-hardcoded-version-1.1.0.tar.gz'
-        assert test_source[0].split()[1] == expected_result
+        assert test_source == expected_result
```

Comment:
Now it doesn't make much sense not to assert directly in the condition. The change also makes `assert test_source` redundant.